azuread/table_azuread_conditional_access_named_location.go

```diff
@@ -0,0 +1,163 @@
+			{Name: "location_info", Type: proto.ColumnType_JSON, Description: "Specifies some location information for the Named Location object. Now supported: IP (v4/6 and CIDR/Range), odata_type, IsTrusted (for IP named locations only). Country (and regions, if exist), lookup method, UnkownCountriesAndRegions (for country named locations only)", Transform: transform.FromMethod("GetLocationInfo")},
+			{Name: "type", Type: proto.ColumnType_STRING, Description: "Specifies the type of the Named Location object: IP or Country", Transform: transform.FromMethod("GetType")},
+			{Name: "created_date_time", Type: proto.ColumnType_TIMESTAMP, Description: "The create date of the Named Location object.", Transform: transform.FromMethod("GetCreatedDateTime")},
+			{Name: "modified_date_time", Type: proto.ColumnType_TIMESTAMP, Description: "The modification date of Named Location object.", Transform: transform.FromMethod("GetModifiedDateTime")},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAdConditionalAccessNamedLocations(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create client
+	client, adapter, err := GetGraphClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("azuread_conditional_access_named_location.listAdConditionalAccessNamedLocations", "connection_error", err)
+		return nil, err
+	}
+
+	// List operations
+	input := &identity.ConditionalAccessNamedLocationsRequestBuilderGetQueryParameters{
+		Top: Int32(1000),
+	}
+
+	limit := d.QueryContext.Limit
+	if limit != nil {
+		if *limit > 0 && *limit < 1000 {
+			l := int32(*limit)
+			input.Top = Int32(l)
+		}
+	}
+
+	equalQuals := d.EqualsQuals
+	filter := buildConditionalAccessNamedLocationQueryFilter(equalQuals)
+
+	if len(filter) > 0 {
+		joinStr := strings.Join(filter, " and ")
+		input.Filter = &joinStr
+	}
+
+	options := &identity.ConditionalAccessNamedLocationsRequestBuilderGetRequestConfiguration{
+		QueryParameters: input,
+	}
+
+	result, err := client.Identity().ConditionalAccess().NamedLocations().Get(ctx, options)
+	if err != nil {
+		errObj := getErrorObject(err)
+		plugin.Logger(ctx).Error("listAdConditionalAccessNamedLocations", "list_conditional_access_named_location_error", errObj)
+		return nil, errObj
+	}
+
+	pageIterator, err := msgraphcore.NewPageIterator[models.NamedLocationable](result, adapter, models.CreateNamedLocationCollectionResponseFromDiscriminatorValue)
+	if err != nil {
+		plugin.Logger(ctx).Error("listAdConditionalAccessNamedLocations", "create_iterator_instance_error", err)
+		return nil, err
+	}
+
+	err = pageIterator.Iterate(ctx, func(pageItem models.NamedLocationable) bool {
+		switch t := pageItem.(type) {
+		case *models.IpNamedLocation:
+			d.StreamListItem(ctx, &ADIpNamedLocationInfo{t})
+		case *models.CountryNamedLocation:
+			d.StreamListItem(ctx, &ADCountryNamedLocationInfo{t})
```

Comment:
I managed to change it, but tool me a lot of time and the results isn't elegant at all. I'm not sure at all this is better than before. If you have a suggestion how to make it better please tell me. I think it might be better before this change.